adapta/storage/distributed_object_store/datastax_astra/_models.py

```diff
@@ -0,0 +1,137 @@
+"""
+ Auxiliary models for Astra client operations.
+"""
+from functools import reduce
+from typing import final, List, Dict, Generic, TypeVar, Any, Union
+
+TField = TypeVar("TField")  # pylint: disable=invalid-name
+
+
+@final
+class AstraFilterExpression:
+    """
+    Filter expression from combining two or more Astra fields
+    """
+
+    def __init__(self, expr: List[Dict[str, Any]]):
+        """
+        Creates an AstraFilterExpression
+        """
+        self._expr = expr
+
+    @property
+    def expression(self) -> List[Dict[str, Any]]:
+        """
+        Generated expression.
+        """
+        return self._expr
+
+    def __and__(self, other: Union["AstraFilterExpression", "AstraField"]) -> "AstraFilterExpression":
+        added_expr = [other.expression] if isinstance(other, AstraField) else other.expression
+        self._expr = reduce(lambda e1, e2: e1 | e2, self._expr + added_expr)
+        return self
+
+    def __or__(self, other: Union["AstraFilterExpression", "AstraField"]) -> "AstraFilterExpression":
+        added_expr = [other.expression] if isinstance(other, AstraField) else other.expression
+        self._expr = self._expr + added_expr
+        return self
+
+
+@final
+class AstraField(Generic[TField]):
```

Comment:
With the current implementation, the following is allowed:

```
a = AstraField("col1")
filter_1 = a == "small boom"
filter_2 = filter_1 == "big boom"
```

Which I don't think makes sense. I suggest that the magic operators return a filter expression instead.